src/app/components/home/search/search.component.ts

```diff
@@ -48,22 +53,19 @@ export class SearchComponent implements OnInit, OnDestroy {
   }
 
   fetchData() {
-    //this is what we use for the actual API call
-    // this.productsService
-    //   .getProducts(this.searchValue)
-    //   .subscribe((productsData) => {
-    //     this.products = productsData;
-    //   });
+    this.store.dispatch(searchProducts({ searchValue: this.searchValue }));
 
-    const prodSub = this.productsService
-      .getProducts(this.searchValue)
-      .subscribe((productsData) => {
-        this.products = productsData.filter((product) =>
+    //this is ugly, remember to remove once backend endpoint is established
+    console.warn(
+      'Remember you have a useless filter function in search.component.ts that needs to be removed.'
+    );
+    this.allProducts$ = this.allProducts$.pipe(
```

Comment:
Do this in the constructor

Reply:
Then `.pipe(takeUntilDestroyed())`

Reply:
this is really just for the JSON data in order to "fake" the search as you type, in reality I'd just be passing the searchValue in the dispatch and the filtered data would be returned, so this whole block will be removed